lib/Backends/Interpreter/InterpreterDeviceManager.cpp

```diff
@@ -108,8 +108,10 @@ void InterpreterDeviceManager::evictNetworkImpl(std::string functionName,
 void InterpreterDeviceManager::runFunctionImpl(
     RunIdentifierTy id, std::string function,
     std::unique_ptr<ExecutionContext> context, ResultCBTy resultCB) {
+  auto ev = context->scopedEvent("DM_run");
```

Comment:
One issue with the scoped events is they're going to include the time taken to call resultCBs if we don't always remember to always call the `end()` method before every callback which may not be easy to maintain the correctness of. If we don't have the scoped thing, it will be obvious if we forget to call the end() method because there will be no matching end() event but with scoped events it will be a bit inaccurate and we won't have any way to know it.

Reply:
I see what you're getting at but a) events will always be wrong if you don't terminate them at the right time, and b) this is just a helper interface, you can still add events manually (such as we do in the LLVMCompiledFunction in the lookupJitmain event.

Are you against using the Scoped version here, or having it at all?